cpp-hello/README.md

```diff
@@ -0,0 +1,259 @@
+Firecracker requires KVM support.
+Xen requires a system with Xen installed.
+
+A successful run will show a message such as the one below:
+
+```text
+Booting from ROM..Powered by
+o.   .o       _ _               __ _
+Oo   Oo  ___ (_) | __ __  __ _ ' _) :_
+oO   oO ' _ `| | |/ /  _)' _` | |_|  _)
+oOo oOO| | | | |   (| | | (_) |  _) :_
+ OoOoO ._, ._:_:_,\_._,  .__,_:_, \___)
+                Calypso 0.17.0~5d38d108
+Hello from Unikraft!
+```
+
+### Run on QEMU/x86_64
+
+```console
+qemu-system-x86_64 -nographic -m 8 -cpu max -kernel out/cpp-hello_qemu-x86_64
+```
+
+### Run on QEMU/ARM64
+
+```console
+qemu-system-aarch64 -nographic -machine virt -m 8 -cpu max -kernel out/cpp-hello_qemu-arm64
+```
+
+### Run on Firecracker/x86_64
+
+```console
+rm -f firecracker.socket
+firecracker-x86_64 --config-file fc.x86_64.json --api-sock firecracker.socket
+```
+
+The user running the above command must be able to use KVM.
+Typically this means being part of the `kvm` group.
+Otherwise, run the command above as root or prefixed by `sudo`.
+
+### Run on Firecracker/ARM64
+
+```console
+rm -f firecracker.socket
+firecracker-aarch64 --config-file fc.arm64.json --api-sock firecracker.socket
+```
+
+The user running the above command must be able to use KVM.
+Typically this means being part of the `kvm` group.
+Otherwise, run the command above as the `root` account or prefixed by `sudo`.
+
+### Run on Xen/x86_64
+
+```console
+sudo xl create -c xen.x86_64.cfg
+```
+
+You need use `sudo` or the `root` account to run Xen.
+
+### Run on Xen/ARM64
+
+```console
+sudo xl create -c xen.arm64.cfg
+```
+
+You need use `sudo` or the `root` account to run Xen.
+
+## Customize
+
+C++ Hello is the simplest application to be run with Unikraft.
+This makes it ideal as a minimal testing ground for new features: it builds fast, it doesn't have dependencies.
+
+### Update the Unikraft Core Code
+
+If updating the Unikraft core code in the `../repos/unikraft/` directory, you then go through the [configure](#configure), [build](#build) and [run](#run) steps.
+
+### Add Other Source Code Files
+
+The current configuration use a single source file: `hello.cpp`.
+If looking to add another file to the build, update the [`Makefile.uk`](Makefile.uk) file.
+
+For example, to add a new file `support.cpp` to the build, update the [`Makefile.uk`](Makefile.uk) file to:
+
+```make
+$(eval $(call addlib,apphelloworldcpp))
+
+APPHELLOWORLDCPP_SRCS-y += $(APPHELLOWORLDCPP_BASE)/hello.cpp
+APPHELLOWORLDCPP_SRCS-y += $(APPHELLOWORLDCPP_BASE)/support.cpp
+```
+
+To add a new include directory, such as a local `include/` directory, update the [`Makefile.uk`](Makefile.uk) file to:
+
+```make
+$(eval $(call addlib,apphelloworldcpp))
+
+APPHELLOWORLDCPP_SRCS-y += $(APPHELLOWORLDCPP_BASE)/hello.cpp
+CINLUDES-y += -I$(APPHELLOWORLDCPP_BASE)/include
+```
+
+Then go through the [configure](#configure), [build](#build) and [run](#run) steps.
+
+### Add Libraries
```

Comment:
Rename this to `Add Other Libraries` and add contents from the [`c-http` application](https://github.com/unikraft/catalog-core/tree/main/c-http#add-other-libraries). You already have a lot of applications present.